emu/src/arm7tdmi.rs

```diff
@@ -168,7 +168,14 @@ impl Arm7tdmi {
                         // bits [11-8] - Shift register (R0-R14) - only lower 8bit 0-255 used
                         let rs = op2.get_bits(8..=11);
 
-                        self.registers.register_at(rs.try_into().unwrap()) & 0xFF
+                        let rs = self.registers.register_at(rs.try_into().unwrap()) & 0xFF;
+
+                        // If shift is taken from register and the value is 0 Rm is directly used as operand
+                        if rs == 0 {
+                            return rm;
+                        }
+
+                        rs
```

Comment:
Can we write a little test about this?

Reply:
Yup, forgot about it. I've added it to the latest commit. Thanks